gammapy/data/tests/test_observations.py

```diff
@@ -38,6 +38,24 @@ def test_data_store_observation(data_store):
     assert_skycoord_allclose(obs.target_radec, c)
 
 
+@requires_data("gammapy-extra")
+def test_create_missing_gti():
+    """Test the DataStoreObservation._create_missing_gti() method.
+
+    For the 'hess-dl3-dr1' data set, the GTI tables contain only the 'TSTART' and 'TSTOP' from the 'event list' and
+    'obs index' header. The `DataStoreObservation._create_missing_gti()` method creates a GTI table on-the-fly using
+    exactly those values.
+    """
+    ds = data_store()
```

Comment:
`data_store` is a session-scoped fixture. Fixtures shouldn't be called, they should be injected into tests. (see other existing uses of this fixture). This will give a warning or error with the latest pytest release.

Suggest to put the `_create_missing_gti` method as `ObservationTable.make_gti_table(obs_id)` and add a test for it there without doing something complex like modifying a DataStore fixture object. Given that this is just a small one-line workaround I think it's fine to have no extra unit test from the `DataStoreObservation` side.